tidy3d/components/medium.py

```diff
@@ -15,6 +15,7 @@
 import numpy as npo
 import pydantic.v1 as pd
 import xarray as xr
+from scipy import signal
```

Comment:
we need to double check with MC because this might add scipy as a "core" dependency, which can slow down the backend. Is it possible to instead do a lazy import of this? see `SimulationData.to_mat_file`

https://github.com/flexcompute/tidy3d/blob/b068ed8d7107fe4381d3013e42d72e74b5a7c2e7/tidy3d/components/data/sim_data.py#L1322-L1329

Reply:
not sure if we want to do exactly this. but maybe double check where we import scipy in other parts of `components/` to see if we have some standard way of doing it. Thanks

Reply:
actually I just did a quick check and it does seem like we use it in a couple places imported from top level, so it's probably fine and we can address all of these when it comes up again.